### PR TITLE
feat(Accident-notification): An/api implementation

### PR DESCRIPTION
**Pull request by Thorrik58**
# ...

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/api/domains/health-insurance-v2/src/lib/api-domains-health-insurance-v2.spec.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { apiDomainsHealthInsuranceV2 } from './api-domains-health-insurance-v2'
+
+describe('apiDomainsHealthInsuranceV2', () => {
+  it('should work', () => {
+    expect(apiDomainsHealthInsuranceV2()).toEqual(
+      'api-domains-health-insurance-v2',
+    )
+  })
+})
```
Comment (eirikurn) [inline libs/api/domains/health-insurance-v2/src/lib/api-domains-health-insurance-v2.spec.ts:9]: Remove?

File: libs/api/domains/health-insurance-v2/src/lib/api-domains-health-insurance-v2.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export function apiDomainsHealthInsuranceV2(): string {
+  return 'api-domains-health-insurance-v2'
+}
```
Comment (eirikurn) [inline libs/api/domains/health-insurance-v2/src/lib/api-domains-health-insurance-v2.ts:3]: Remove?

File: libs/api/domains/health-insurance-v2/src/lib/graphql/health-insurance-v2.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+@Resolver(() => AccidentNotificationStatus)
+export class HealthInsuranceResolver {
+  constructor(
+    private accidentNotificationApi: HealthInsuranceService,
+    @Inject(LOGGER_PROVIDER)
+    private logger: Logger,
+  ) { }
+
+  @Query(() => AccidentNotificationStatus, {
+    name: 'accidentStatus',
```
Comment (eirikurn) [inline libs/api/domains/health-insurance-v2/src/lib/graphql/health-insurance-v2.resolver.ts:21]: Should follow the [graphql naming convention](https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions#integrating-naming-conventions-into-shared-api).

File: libs/api/domains/health-insurance-v2/src/lib/graphql/dto/AccidentStatus.input.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { Field, InputType } from '@nestjs/graphql'
+import { IsNumber } from 'class-validator'
+
+@InputType()
+export class AccidentStatusInput {
```
Comment (eirikurn) [inline libs/api/domains/health-insurance-v2/src/lib/graphql/dto/AccidentStatus.input.ts:5]: Should follow the [graphql naming convention](https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions#integrating-naming-conventions-into-shared-api).

File: libs/api/domains/health-insurance-v2/src/lib/graphql/health-insurance-v2.service.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  constructor(
+    private readonly accidentNotificationApi: DocumentApi,
+    @Inject(LOGGER_PROVIDER)
+    private logger: Logger,
+  ) { }
+
+  async getAccidentNotificationStatus(
+    ihiDocumentID: number,
+  ): Promise<AccidentNotificationStatus | null> {
+    console.log('starting call to get accident', ihiDocumentID)
```
Comment (eirikurn) [inline libs/api/domains/health-insurance-v2/src/lib/graphql/health-insurance-v2.service.ts:68]: Should use logger.

File: libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationAttachment.model.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { Field, ObjectType } from '@nestjs/graphql'
+import { IsEnum } from 'class-validator'
+
+export enum AttachmentTypes {
```
Comment (eirikurn) [inline libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationAttachment.model.ts:4]: Should follow the [graphql naming convention](https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions#integrating-naming-conventions-into-shared-api).

File: libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationAttachment.model.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  INJURY_CERTIFICATE = 'InjuryCertificate',
+  PROXY_DOCUMENT = 'ProxyDocument',
+  POLICE_REPORT = 'PoliceReport',
+}
+
+@ObjectType()
+export class AccidentNotificationAttachment {
```
Comment (eirikurn) [inline libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationAttachment.model.ts:11]: Should follow the [graphql naming convention](https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions#integrating-naming-conventions-into-shared-api).

File: libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationConfirmation.model.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import { Field, ObjectType } from '@nestjs/graphql'
+import { IsEnum } from 'class-validator'
+
+export enum ConfirmationTypes {
```
Comment (eirikurn) [inline libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationConfirmation.model.ts:4]: Should follow the [graphql naming convention](https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions#integrating-naming-conventions-into-shared-api).

File: libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationConfirmation.model.ts [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+  INJUREDORREPRESENTATIVEPARTY = 'InjuredOrRepresentativeParty',
+  COMPANYPARTY = 'CompanyParty',
+}
+
+@ObjectType()
+export class AccidentNotificationConfirmation {
```
Comment (eirikurn) [inline libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationConfirmation.model.ts:10]: Should follow the [graphql naming convention](https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions#integrating-naming-conventions-into-shared-api).

File: libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationStatus.model.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
+import { Field, ObjectType } from '@nestjs/graphql'
+import { AccidentNotificationAttachment } from './accidentNotificationAttachment.model'
+import { AccidentNotificationConfirmation } from './accidentNotificationConfirmation.model'
+import { IsEnum } from 'class-validator'
+
+export enum StatusTypes {
```
Comment (eirikurn) [inline libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationStatus.model.ts:6]: Should follow the [graphql naming convention](https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions#integrating-naming-conventions-into-shared-api).

File: libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationStatus.model.ts [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+  ACCEPTED = 'Accepted',
+  REFUSED = 'Refused',
+  INPROGRESS = 'InProgress',
+  INPROGRESSWAITINGFORDOCUMENT = 'InProgressWaitingForAttachment',
+}
+
+@ObjectType()
+export class AccidentNotificationStatus {
```
Comment (eirikurn) [inline libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationStatus.model.ts:14]: Should follow the [graphql naming convention](https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions#integrating-naming-conventions-into-shared-api).

File: libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationConfirmation.model.ts [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+  @Field(() => Boolean)
+  isReceived?: boolean
+
+  // 1 = Injured or representative party, 2 = Company party
+  @Field(() => String)
```
Comment (eirikurn) [inline libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationConfirmation.model.ts:15]: Use GraphQL enum?

File: libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationAttachment.model.ts [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+  @Field(() => Boolean)
+  isReceived?: boolean
+
+  // 1 = Áverkavottorð, 2 = Umboðsskjal, 3 = Lögregluskýrsla
+  @Field(() => String)
+  @IsEnum(AttachmentTypes)
```
Comment (eirikurn) [inline libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationAttachment.model.ts:17]: Use GraphQL enum?

File: libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationStatus.model.ts [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+  @Field(() => Number)
+  numberIHI?: number
+
+  @Field(() => String)
+  @IsEnum(StatusTypes)
+  status?: string
```
Comment (eirikurn) [inline libs/api/domains/health-insurance-v2/src/lib/graphql/models/accidentNotificationStatus.model.ts:20]: Use GraphQL enum?

File: libs/api/domains/health-insurance-v2/src/lib/health-insurance-v2-module.ts [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+@Module({})
+export class HealthInsuranceV2Module {
+  static register(options: HealthInsuranceServiceOptions): DynamicModule {
+    return {
+      module: HealthInsuranceV2Module,
+      imports: [HealthInsuranceV2Client.register(options)],
+      providers: [HealthInsuranceResolver, HealthInsuranceService],
+    }
+  }
+}
```
Comment (eirikurn) [inline libs/api/domains/health-insurance-v2/src/lib/health-insurance-v2-module.ts:18]: Should add these new queries/models/clients to the existing health insurance domain?

File: libs/clients/health-insurance-v2/src/lib/clients-health-insurance-v2.module.ts [hunk 16]
```diff
@@ -0,0 +1,46 @@
+import { DynamicModule } from '@nestjs/common'
+import { Configuration, DocumentApi } from '../../gen/fetch'
+import { HealthInsuranceV2Options } from './clients-health-insurance-v2.config'
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+
+export class HealthInsuranceV2Client {
```
Comment (eirikurn) [inline libs/clients/health-insurance-v2/src/lib/clients-health-insurance-v2.module.ts:6]: Is this API a superset of (or meant to replace) the existing health-insurance APIs?

**Review by eirikurn — CHANGES_REQUESTED** [review]
Only reviewed api domain and client. 👍 

Interested in how this relates to existing health insurance APIs and functionality.

File: libs/clients/health-insurance-v2/src/lib/clients-health-insurance-v2.module.ts (hunk 16 above)
Comment (Thorrik58) [reply libs/clients/health-insurance-v2/src/lib/clients-health-insurance-v2.module.ts:6]: The plan is to replace the existing one once the sjukra accident notification application has been delivered.

File: libs/api/domains/health-insurance/src/lib/graphql/scalars.ts [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+  name: 'HealthInsuranceAccidentNotificationConfirmationTypes',
+})
+
+registerEnumType(HealthInsuranceAccidentNotificationStatusTypes, {
+  name: 'HealthInsuranceAccidentNotificationStatusTypes',
+})
+
+registerEnumType(HealthInsuranceAccidentNotificationAttachmentTypes, {
+  name: 'HealthInsuranceAccidentNotificationAttachmentTypes',
+})
```
Comment (eirikurn) [inline libs/api/domains/health-insurance/src/lib/graphql/scalars.ts:14]: Fine to have a simpler name in TypeScript and just register the long name in a GraphQL decorator and/or registerEnumType. :)

**Review by eirikurn — APPROVED** [review]
Thanks for merging the domains! This is much nicer.

I look forward to getting rid of the SOAP API client 😄

File: libs/clients/health-insurance-v2/src/lib/clients-health-insurance-v2.module.ts (hunk 16 above)
Comment (eirikurn) [reply libs/clients/health-insurance-v2/src/lib/clients-health-insurance-v2.module.ts:6]: Can't wait! :D

**Review by obmagnusson — APPROVED** [review]
LGTM

**Pull request merged**